modules/listItems.js

```diff
@@ -1,16 +1,32 @@
 import db from './db';
-import { populateItems } from './populateItems';
+import {
+  populateItems,
+  selectedSection,
+  setSelectedSection,
+} from './populateItems';
 
-// actions available on the list items themselves
-// actions on the list items themselves
-// TODO: figure out why this module isn't working - I'm suspecting something with scope
-// try to use in index.js and you'll get an error that db and populateItems can't be found...
+/** Toggle Purchase (in cart) Status - items will appear crossed off */
 export const toggleItemPurchaseStatus = async (event, id) => {
   await db.items.update(id, { isPurchased: !!event.target.checked });
   await populateItems();
 };
 
+/** Remove a Shopping List item */
 export const removeItem = async (id) => {
+  if (selectedSection) {
+    console.log('removeItems clause hit');
+    const itemToRemove = await db.items.get(id);
+    // get the count of total items in the section
+    const itemsInSection = await db.items
+      .where('section')
+      .equals(itemToRemove.section)
+      .count();
+    // if there's only this one, reset the selectedSection
+    if (itemsInSection === 1 && itemToRemove.section === selectedSection) {
+      setSelectedSection(null);
+    }
+  }
+
```

Comment:
These weren't even being properly used before 🤦 Which is just a sign that I need to clean things ups here.